Auto mTLS status for Stable. (#35)
istio/enhancements: features/auto_mtls.md

**Bugs**

- [X] Feature has no known major issues.

**Approvals**

- [X] The appropriate work group(s) have reviewed and approved promotion of the [feature](https://docs.google.com/document/d/1cU4TFqYdir1luUN4gf3mTvujb4LugNLhcNffOluQyns/edit?disco=AAAALq26fXg&ts=60467f2f&usp_dm=true&resourcekey=0-tIc8kq3LHGokG4FHfXJ5CQ).
- [ ] The supportability review panel has reviewed promotion of the feature. TODO: WIP.  

> - [X] The TOC has reviewed and approved promotion of the feature as part of the

Reviewer: You can also refer to Istio 2021 roadmap (https://docs.google.com/spreadsheets/d/1FJ3AJFFnGw6TxZAoyj-zgFwTn37WsMKTOGNR4FjG7T8/edit#gid=117171340) which has been reviewed and approved by TOC.

**Approvals**

- [X] The appropriate work group(s) have reviewed and approved promotion of the [feature](https://docs.google.com/document/d/1cU4TFqYdir1luUN4gf3mTvujb4LugNLhcNffOluQyns/edit?disco=AAAALq26fXg&ts=60467f2f&usp_dm=true&resourcekey=0-tIc8kq3LHGokG4FHfXJ5CQ).
- [ ] The supportability review panel has reviewed promotion of the feature.  

> - [X] The TOC has reviewed and approved promotion of the feature as part of the

Reviewer 2: @brian-avery I think this section of the doc is confusing/doesn't match my expectations. I would expect that this section is "TOC approves promotion to stable [as a result of meeting all expected criteria and agreeing its met the bar]". However, it seems to instead say "TOC approved that we should try to promote it this release" - which means we have no checkoff to assert that we actually met all the criteria we wanted.

Sorry to hijack your PR @incfly

Reviewer 3: Yeah, interesting comment:
* This expects that feature promotions are understood and reflected in the scope of work expected by a work group in the release during the roadmap review so that users, work groups, and developers can plan accordingly. This is something that came out of the definition of done. Perhaps a bit of scope creep for the checklist, but that's the idea behind it. 
* Final sign-off by TOC is actually inherent to the process at this point in that TOC must approve the update to features.yaml which is the final signoff on promotion.

Author: At the first sight, I feel presenting to TOC and have it approved the status matters. But thinking more, if such feature template is complete and sufficient, all checklist are done, PRs are reviewed by right reviewers, I am not sure what we gain by discussing in TOC meeting? Personally,  PR reviewing and formalizing as checklist seems much a better standard procedure.

I agree we might need to ensure TOC's representative (or representatives from other aspects) in the PR review for feature promotion process.

> - [X] The TOC has reviewed and approved promotion of the feature as part of the

Reviewer 3: Does the following make it more clear? If so, we can update the template

```suggestion
- [X] Reviewed in a TOC meeting as part of the roadmap for a release.
```
